### PR TITLE
Add team resource for Sysdig Secure

### PR DESCRIPTION
**Pull request by rung**
### What
- Add `sysdig_secure_team` resource

### Why
- This PR is related to #11 

### Comment for reviewers
- I think we have additional discussion topic about this PR.
  - So I'll write comment on the point which I want reviewers to check.

### Test result
<details><summary>make testacc</summary><div>

```bash
$ make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v  -timeout 120m
?   	github.com/draios/terraform-provider-sysdig	[no test files]
?   	github.com/draios/terraform-provider-sysdig/scripts/oanc	[no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccAlertAnomaly
=== PAUSE TestAccAlertAnomaly
=== RUN   TestAccAlertDowntime
=== PAUSE TestAccAlertDowntime
=== RUN   TestAccAlertEvent
=== PAUSE TestAccAlertEvent
=== RUN   TestAccAlertGroupOutlier
=== PAUSE TestAccAlertGroupOutlier
=== RUN   TestAccAlertMetric
=== PAUSE TestAccAlertMetric
=== RUN   TestAccNotificationChannel
=== PAUSE TestAccNotificationChannel
=== RUN   TestAccPolicy
=== PAUSE TestAccPolicy
=== RUN   TestAccRuleContainer
--- PASS: TestAccRuleContainer (4.32s)
=== RUN   TestAccRuleFalco
--- PASS: TestAccRuleFalco (2.64s)
=== RUN   TestAccRuleFilesystem
--- PASS: TestAccRuleFilesystem (5.11s)
=== RUN   TestAccRuleNetwork
--- PASS: TestAccRuleNetwork (4.99s)
=== RUN   TestAccRuleProcess
--- PASS: TestAccRuleProcess (2.82s)
=== RUN   TestAccRuleSyscall
--- PASS: TestAccRuleSyscall (2.80s)
=== RUN   TestAccTeam
--- PASS: TestAccTeam (7.42s)
=== RUN   TestAccUser
--- PASS: TestAccUser (4.91s)
=== CONT  TestAccAlertAnomaly
=== CONT  TestAccAlertMetric
=== CONT  TestAccAlertEvent
=== CONT  TestAccAlertGroupOutlier
=== CONT  TestAccAlertDowntime
=== CONT  TestAccPolicy
=== CONT  TestAccNotificationChannel
--- PASS: TestAccAlertDowntime (1.70s)
--- PASS: TestAccAlertEvent (1.71s)
--- PASS: TestAccAlertAnomaly (1.73s)
--- PASS: TestAccAlertGroupOutlier (1.73s)
--- PASS: TestAccAlertMetric (2.53s)
--- PASS: TestAccNotificationChannel (4.45s)
--- PASS: TestAccPolicy (7.22s)
PASS
ok  	github.com/draios/terraform-provider-sysdig/sysdig	44.176s
?   	github.com/draios/terraform-provider-sysdig/sysdig/monitor	[no test files]
?   	github.com/draios/terraform-provider-sysdig/sysdig/secure	[no test files]
```

</div></details>

<details><summary>make test</summary><div>

```bash
$ make test
==> Checking that code complies with gofmt requirements...
go test ./... -timeout=30s -parallel=4
?   	github.com/draios/terraform-provider-sysdig	[no test files]
?   	github.com/draios/terraform-provider-sysdig/scripts/oanc	[no test files]
ok  	github.com/draios/terraform-provider-sysdig/sysdig	0.523s
?   	github.com/draios/terraform-provider-sysdig/sysdig/monitor	[no test files]
?   	github.com/draios/terraform-provider-sysdig/sysdig/secure	[no test files]
PC2292:terraform-provider-sysdig suezawa$
```

</div></details>

### CODE REVIEW

File: sysdig/resource_sysdig_secure_team.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"theme": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "#73A1F7",
```
Comment (rung) [inline sysdig/resource_sysdig_secure_team.go]: #73A1F7 is this color
image

File: sysdig/resource_sysdig_secure_team.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+					Schema: map[string]*schema.Schema{
+						"email": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"role": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "ROLE_TEAM_STANDARD",
```
Comment (rung) [inline sysdig/resource_sysdig_secure_team.go]: I set "ROLE_TEAM_STANDARD" as default role. is it ok?
https://docs.sysdig.com/en/user-and-team-administration.html#UUID-b7b6f0fb-85ec-5366-e3e4-69260be044fd_UUID-87f1d5be-77eb-4f1b-51d9-15612ba9d289

File: sysdig/resource_sysdig_secure_team.go (hunk 1 above)
Comment (tembleking) [reply sysdig/resource_sysdig_secure_team.go]: Perfect!

File: sysdig/resource_sysdig_secure_team.go (hunk 2 above)
Comment (tembleking) [reply sysdig/resource_sysdig_secure_team.go]: Yes, it's correct :slightly_smiling_face:

**Comment by tembleking** [comment]
Thanks again @rung!

File: sysdig/secure/teams.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
+package secure
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+func (client *sysdigSecureClient) getUserIdbyEmail(userRoles []UserRoles) ([]UserRoles, error) {
```
Comment (rung) [inline sysdig/secure/teams.go]: Sysdig API(/api/teams) requires User ID as identifier(not email address).
But I use email address as identifier on `sysdig_secure_team` resource.

This func gets User ID using Sysdig API(/api/users/light).

Comment (tembleking) [reply sysdig/secure/teams.go]: Perfect! We can use it to implement a Data Source for Users :slightly_smiling_face:

**Comment by rung** [comment]
@tembleking Thank you for your review!

**Pull request merged**